app/com/linkedin/drelephant/clients/azkaban/AzkabanJobStatusUtil.java

```diff
@@ -29,37 +35,57 @@
  * This class is azkaban scheduler util for getting job status.
  */
 public class AzkabanJobStatusUtil {
-  private AzkabanWorkflowClient _workflowClient;
+
+  private static final Logger logger = Logger.getLogger(AzkabanJobStatusUtil.class);
+  private HashMap<String, AzkabanWorkflowClient> workflowClients = new HashMap<String, AzkabanWorkflowClient>();
   private String scheduler = "azkaban";
   private static String USERNAME = "username";
   private static String PRIVATE_KEY = "private_key";
   private static String PASSWORD = "password";
+  private static final long TOKEN_UPDATE_INTERVAL = AutoTuner.ONE_MIN * 60 * 1;
 
-  /**
-   * Constructor of the class
-   * @param url
-   */
-  public AzkabanJobStatusUtil(String url) {
```

Comment:
I think this class can be marked as a singleton?

Reply:
This class is not thread safe, whoever is using this class need to make sure that it runs only one request at a time. So we need to first make it thread safe or make it a utility kind of class before making it singleton.

Reply:
I am not happy with keeping the login state in a utility method. Ideally, the login should be done in a separate class (workflowXlient itself) and then you can call this utility method by passing the workflowClient object to it. This way the login can be reused for other utilities and the utility methods also remain stateless and easy testable.

Reply:
Currently also login method is called form outside. As we discussed we will create a task to refactor this and take it up with in one month.